### PR TITLE
Add lps22hb dev-board sensor

### PR DESCRIPTION
**Pull request by madninja**
Adds the lps22hb lua library and datasheet

### CODE REVIEW

File: script/lps22hb/lps22hb.lua [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    status, reason = o:is_connected()
+    if not status then
+        return status, reason
+    end
+    return o
+end
+
+
+function lps22hb:is_connected()
+    local status, result = self:_get(self.ID, "B")
```
Comment (Vagabond) [inline script/lps22hb/lps22hb.lua]: This is too magical, add some comments?

File: script/lps22hb/lps22hb.lua [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+    PRESS_OUT_XL = 0x28,
+    TEMP_OUT_L   = 0xAB
+}
+
+
+function lps22hb:new(address)
+    address = address or lps22hb.DEFAULT_ADDRESS
+    local o = { address = address }
+    setmetatable(o, self)
```
Comment (Vagabond) [inline script/lps22hb/lps22hb.lua]: explain what is going on here

File: script/lps22hb/lps22hb.lua [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    local status, buffer = i2c.txn(i2c.tx(self.address, reg),
+                                   i2c.rx(self.address, pack_size))
+    if not status then
+        return false, "failed to get value from device"
+    end
+    -- get the values and capture them in a list
+    values = {string.unpack(pack_fmt, buffer)}
+    -- call conversion function if given
+    if convert then
+        values = {convert(table.unpack(values))}
```
Comment (Vagabond) [inline script/lps22hb/lps22hb.lua]: you can't map the list here? seems like you might make extra garbage initializing the table twice

File: script/lps22hb/lps22hb.lua [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+    end
+    -- return the values as an unpacked tuple
+    return true, table.unpack(values)
+end
+
+
+function lps22hb:_pressure_get()
+    local status, result = self:_get(self.PRESS_OUT_XL, "i3")
```
Comment (Vagabond) [inline script/lps22hb/lps22hb.lua]: Should we consistently use the endianness marker?

File: script/lps22hb/lps22hb.lua [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+
+function lps22hb:_oneshot_read(func)
+    -- request one shot reading
+    local status =
+        i2c.txn(i2c.tx(self.address, self.CTRL_REG2, self.CTRL_REG2_ONESHOT))
+    if not status then
+        return false, "failed to start reading"
+    end
+
+    -- now loop reading the result register waiting for a response
```
Comment (Vagabond) [inline script/lps22hb/lps22hb.lua]: This seems pretty terrible, this is really what you have to do?

File: script/lps22hb/lps22hb.lua [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+
+
+-- Turn on power
+he.power_set(true)
+
+sensor = assert(lps22hb:new())
+
+local now = he.now()
+while true do
+    local _, pressure = assert(sensor:read_pressure())
```
Comment (Vagabond) [inline script/lps22hb/lps22hb.lua]: Maybe for 'success', given the limited scope of 'falsish' things in Lua, don't return the `true, value`, just return the value?

File: script/lps22hb/lps22hb.lua [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+    local _, temperature = assert(sensor:read_temperature())
+
+    he.power_set(false)
+    he.send("t", now, "f", temperature)
+    he.send("p", now, "f", pressure)
+
+    -- Un-comment the following line to see results in semi-hosted mode
+    -- print(temperature, pressure)
+
+    now = he.wait{time=now + SAMPLE_INTERVAL}
```
Comment (Vagabond) [inline script/lps22hb/lps22hb.lua]: How is `now` incrementing here? it only seems to be modified outside the loop?

File: script/lps22hb/lps22hb.lua [hunk 8]
```diff
@@ -0,0 +1,128 @@
+-- Development board on-board lps22hb sensor script
+
+i2c = he.i2c
+
+SAMPLE_INTERVAL = 5000 -- 1 minute
+
+lps22hb = {
+    DEFAULT_ADDRESS = 0x5C,
```
Comment (Vagabond) [inline script/lps22hb/lps22hb.lua]: Add more datasheet references here.

**Review by Vagabond — CHANGES_REQUESTED** [review]

File: script/lps22hb/lps22hb.lua (hunk 3 above)
Comment (madninja) [reply script/lps22hb/lps22hb.lua]: most usecases do not use a convert function.. doing the other way means declaring basically an `id` convert function that would be executed if no convert was passed? That would make it simpler at the expense of calling a no-op function every time

File: script/lps22hb/lps22hb.lua (hunk 5 above)
Comment (madninja) [reply script/lps22hb/lps22hb.lua]: @Vagabond yup

File: script/lps22hb/lps22hb.lua (hunk 7 above)
Comment (madninja) [reply script/lps22hb/lps22hb.lua]: he.wait returns the current system time on return. Documented here: https://dev.helium.com/script-api/base/#sleeping

File: script/lps22hb/lps22hb.lua (hunk 8 above)
Comment (madninja) [reply script/lps22hb/lps22hb.lua]: Fixed in 	29a8521

File: script/lps22hb/lps22hb.lua (hunk 2 above)
Comment (madninja) [reply script/lps22hb/lps22hb.lua]: Fixed in 	29a8521

File: script/lps22hb/lps22hb.lua (hunk 1 above)
Comment (madninja) [reply script/lps22hb/lps22hb.lua]: Fixed in 29a8521

File: script/lps22hb/lps22hb.lua (hunk 3 above)
Comment (madninja) [reply script/lps22hb/lps22hb.lua]: Done per our discussion.. Fixed in 29a8521

File: script/lps22hb/lps22hb.lua (hunk 4 above)
Comment (madninja) [reply script/lps22hb/lps22hb.lua]: Fixed in 29a8521

File: script/lps22hb/lps22hb.lua (hunk 6 above)
Comment (madninja) [reply script/lps22hb/lps22hb.lua]: Fixed in 29a8521

File: script/lps22hb/lps22hb.lua [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+
+function lps22hb:new(address)
+    address = address or lps22hb.DEFAULT_ADDRESS
+    -- We use a simple lua object system as defined
+    -- https://www.lua.org/pil/16.1.html
+    -- construct the object table
+    local o = { address = address }
+    -- ensure that the "class" is the main table
+    setmetatable(o, self)
+    -- to ensure none ":" based calls also work
```
Comment (Vagabond) [inline script/lps22hb/lps22hb.lua]: Not sure what this means?

Comment (madninja) [reply script/lps22hb/lps22hb.lua]: Attempted slightly more linkage in 118d0c0

**Review by Vagabond — APPROVED** [review]

**Pull request merged**